### ISSUE TITLE
Lost Connection support

### ISSUE THREAD

**Issue by kevindoveton** [jswetzen/ppstagedisplay#12]
Following Issue #11, add support for a lost connection - i'm happy to take this @jswetzen

**Comment by jswetzen** [comment]
I worked long and hard with this problem and never found a (node js) solution. The problem is that there's no way to tell if the connection it lost in case wifi goes down. I'm pretty sure it handles a disconnected ProPresenter, unless a new version has changed things. But the only way I found to fix a wifi loss was in Linux, by checking if the network connection is down and in that case restart the node process. That solution might be present in the Raspberry Pi image in the only available release.

The interesting thing about the error is that it first got a strange message (DispayLayout without an s), but i don't think that caused the error directly. I managed to parse that message just fine, so I think it was some other message (or rubbish data because of the network loss) following it.

It may be that communication is lost before that error and then there's no reason to try to recover from it.

**Comment by jswetzen** [comment]
Maybe catching this error is enough for it to connect again, but it would be good to log something whenever the messsage is undefined.

I spent hours debugging this kind of problem when writing reconnector.js, so it would be good to manually test this fix a bit too. Seeing how it reacts when ProPresenter is restarted and wifi is lost in different combinations (together with logging) is a good start.

**Comment by jswetzen** [comment]
And thanks again for taking this on :)

**Comment by kevindoveton** [comment]
I've had a bit more of a think about this, I think the easiest solution would be to attempt to retry for 'x' attempts and then to give up and start searching via bonjour. 

If propresenter closes / crashes it gets assigned a new port and so there is no use trying to reconnect forever.

**Comment by jswetzen** [comment]
That's true. I think the clients should be able to stay connected even if the ProPresenter connection was completely reset.
